spec/openapi.yaml

```diff
@@ -85,150 +365,257 @@ paths:
       responses:
         200:
           description: Added new resource
-        404:
-          description: The specified resource was not found
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Error'
+                $ref: '#/components/schemas/Resource'
+        401:
+          $ref: '#/components/responses/Unauthorized'
       security:
       - ApiKeyAuth: []
   /search:
     get:
-      description: Searches the API for a given keyword
-      responses:
-        200: {}
-  /apikey:
-    post:
-      description: Request API key by providing Operation Code credentials
+      summary: SEARCH FOR RESOURCES BY KEYWORD
+      description: Searches for resources containing a given keyword. For example, to search for any resource containing the word `algorithm`, make a `GET` request to `search?q=algorithm`
+      parameters:
+        - in: query
+          name: q
+          schema:
+            type: string
+          required: true
       responses:
         200:
-          description: API Key response
+          description: Search results returned
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/ApiKeyResponse'
-        401:
-          description: Unauthorized
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Error'
+                $ref: '#/components/schemas/SearchResponse'
+        404:
+          $ref: '#/components/responses/NotFound'
+  
 components:
   schemas:
     Error:
-      required:
-      - code
-      - message
       type: object
       properties:
-        code:
+        apiVersion: 
           type: string
-        message:
+        data: 
+          type: object
+        errors: 
           type: string
+        status: 
+          type: string
+        status_code: 
+          type: integer
+
     Resource:
       type: object
       properties:
         apiVersion:
           type: string
-        status_code:
-          type: integer
         data:
           $ref: '#/components/schemas/Resource_data'
+        status_code:
+          type: integer
         status:
           type: string
-    ApiKeyResponse:
+    Category:
       type: object
       properties:
         apiVersion:
           type: string
         data:
-          $ref: '#/components/schemas/ApiKeyResponse_data'
+          $ref: '#/components/schemas/Category_data'
+        has_next:
+          type: integer
+        has_previous:
+          type: integer
+        number_of_pages:
+          type: integer
```

Comment:
This also defaults to `0`, we should have good example defaults for this schema as well as the lists for Languages and Categories